devtools-ci-index.yaml

```diff
@@ -827,10 +837,20 @@
                 sleep 60
             done
             echo 'Using Host' $CICO_hostname
-            set -x
 
-            env > jenkins-env
-            cat $creds_config_file >> jenkins-env
+            (
+                # run in subshell not to affect current environment
+                set +x
+
+                # load creds_config_file
+                eval "$(sed '/^\s*$/d;s/^/export /' $creds_config_file)"
+
+                cp ~/cico-tools/env-toolkit .
+                ./env-toolkit dump -f jenkins-env.json
+                env > jenkins-env
+            )
+
+            set -x
```

Comment:
This is the most complicated thing to understand. Moving from:

```bash
env > jenkins-env
cat $creds_config_file >> jenkins-env
```

to:

```bash
(
    # run in subshell not to affect current environment
    set +x

    # load creds_config_file
    eval "$(sed '/^\s*$/d;s/^/export /' $creds_config_file)"

    cp ~/cico-tools/env-toolkit .
    ./env-toolkit dump -f jenkins-env.json
    env > jenkins-env
)

```

The piece is being executed in a subshell `(...)` because it's modifying the environment by running `eval`, and this prevents from it changing the outside environment.

The `sed` is there to clean the empty lines: `/^\s*$/d` and to prepend `export `. The `$creds_config_file` has a `KEY="value"` format with multiple lines and some empty lines, for example:

```bash
KEY1="val1 blabla"
KEY2="val2"

KEY3="val3"

Reply:
Also the it's doing `eval "$(command ...)"` because it's the best way to ensure it respects spaces, etc. The double-quotes are important.